File: akka-sample-cqrs-java/src/main/java/sample/cqrs/EventProcessorSettings.java
package sample.cqrs;

import akka.actor.typed.ActorSystem;
import com.typesafe.config.Config;

public class EventProcessorSettings {

  public static EventProcessorSettings create(ActorSystem<?> system) {
    return create(system.settings().config().getConfig("event-processor"));
  }

  public static EventProcessorSettings create(Config config) {
    return new EventProcessorSettings(
      config.getString("tag-prefix"),
      config.getInt("parallelism")
    );
  }

  public final String tagPrefix;
  public final int parallelism;

  public EventProcessorSettings(String tagPrefix, int parallelism) {
    this.tagPrefix = tagPrefix;
    this.parallelism = parallelism;
  }
}

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessorSettings.scala
package sample.cqrs

import akka.actor.typed.ActorSystem
import com.typesafe.config.Config

object EventProcessorSettings {

  def apply(system: ActorSystem[_]): EventProcessorSettings = {
    apply(system.settings.config.getConfig("event-processor"))
  }

  def apply(config: Config): EventProcessorSettings = {
    val tagPrefix: String = config.getString("tag-prefix")
    val parallelism: Int = config.getInt("parallelism")
    EventProcessorSettings(tagPrefix, parallelism)
  }
}

final case class EventProcessorSettings(tagPrefix: String, parallelism: Int)
